src/v/rpc/transport.cc

```diff
@@ -281,76 +281,79 @@ transport::do_send(sequence_t seq, netbuf b, rpc::client_opts opts) {
-              // seg. fault.
-              //
-              // And if a scheduling point is added after
-              // `_requests_queue.erase` the conditional for executing the
-              // lambda could succeed for two different messages concurrently
-              // resulting in incorrect ordering of the sent messages.
-              [this, trigger_seq] {
-                  auto it = _requests_queue.begin();
-                  _last_seq = it->first;
-                  auto v = std::move(*it->second->scattered_message);
-                  auto corr = it->second->correlation_id;
-                  _requests_queue.erase(it);
-
-                  auto resp_it = _correlations.find(corr);
-                  if (resp_it == _correlations.end()) {
-                      // request had already completed even before we sent it
-                      // (probably due to timeout or disconnect). We don't need
-                      // to do anything.
-                      return ss::now();
-                  }
-                  auto& resp_entry = resp_it->second;
-
-                  // These units are released once we are out of scope here
-                  // and that is intentional because the underlying write call
-                  // to the batched output stream guarantees us the in-order
-                  // delivery of the dispatched write calls, which is the intent
-                  // of holding on to the units up until this point.
-                  auto units = std::move(resp_entry->resource_units);
-                  auto msg_size = v.size();
-
-                  auto f = _out.write(std::move(v));
-                  resp_entry->timing.dispatched_at = clock_type::now();
-                  vlog(
-                    rpclog.trace,
-                    "Dispatched request with sequence: {}, trigger sequence: "
-                    "{}, "
-                    "correlation_idx: {}, "
-                    "pending queue_size: {}, target_address: {}",
-                    _last_seq,
-                    trigger_seq,
-                    corr,
-                    _requests_queue.size(),
-                    server_address());
-                  return std::move(f)
-                    .then([this, corr](bool flushed) {
-                        if (auto maybe_timing = get_timing(corr)) {
-                            maybe_timing->written_at = clock_type::now();
-                            maybe_timing->flushed = flushed;
-                        }
-                    })
-                    .finally(
-                      [this, msg_size] { _probe.add_bytes_sent(msg_size); });
-              });
-        }).handle_exception([this](std::exception_ptr e) {
-            vlog(rpclog.info, "Error dispatching socket write:{}", e);
-            _probe.request_error();
-            fail_outstanding_futures();
-        });
+      = ssx::spawn_with_gate_then(
+          _dispatch_gate,
+          [this, trigger_seq]() mutable {
```

Comment:
can you split the change from the formatting? no idea what changed here

Reply:
This is a formatting only change of previous commit..

```
[clang-format] rpc/transport: Additional debug information on dispatches
```

Reply:
oh ok, thanks. i didn't notice the [clang-format] thing